### PR TITLE
FSET-1094: Edit candidate's application details (admin)

### PR DESCRIPTION
**Pull request by Lansalo**

### CODE REVIEW

File: app/services/application/ApplicationService.scala [hunk 1]
```diff
@@ -49,4 +53,29 @@ trait ApplicationService {
...
+            )) ::
+            contactDetailsRepository.update(userId, currentCd.copy(
+              outsideUk = editRequest.outsideUk.getOrElse(editRequest.country.isDefined),
+              address = editRequest.address,
+              postCode = editRequest.postCode,
+              country = editRequest.country,
+              phone = editRequest.phone
+            )) :: Nil
+      )
+    } yield ()
```
Comment (benjstephenson) [inline app/services/application/ApplicationService.scala]: we should probably trigger an audit event for this update

File: test/services/application/ApplicationServiceSpec.scala [hunk 2]
```diff
@@ -62,12 +66,53 @@ class ApplicationServiceSpec extends PlaySpec with BeforeAndAfterEach with Mocki
...
+
+      verify(contactDetailsRepositoryMock).find(eqTo(userId))
+      verify(personalDetailsRepositoryMock).find(applicationId)
+
+      verify(personalDetailsRepositoryMock).update(eqTo(applicationId), eqTo(userId), eqTo(expectedPersonalDetails))
+      verify(contactDetailsRepositoryMock).update(eqTo(userId), eqTo(expectedContactDetails))
+      verifyNoMoreInteractions(personalDetailsRepositoryMock, contactDetailsRepositoryMock)
+    }
+
+    "edit candidate shout stop executing if an operation fails" in new ApplicationServiceFixture {
```
Comment (benjstephenson) [inline test/services/application/ApplicationServiceSpec.scala]: typo?

File: app/services/application/ApplicationService.scala (hunk 1 above)
Comment (henricook) [inline app/services/application/ApplicationService.scala]: Definitely

Comment (Lansalo) [inline app/services/application/ApplicationService.scala]: Added. But is not part of the story.

File: test/services/application/ApplicationServiceSpec.scala (hunk 2 above)
Comment (Lansalo) [inline test/services/application/ApplicationServiceSpec.scala]: Typo and repetition. Done.

File: app/services/application/ApplicationService.scala (hunk 1 above)
Comment (henricook) [inline app/services/application/ApplicationService.scala]: Audit of any change in state is a Non Functional Requirement - so applies to everything we do

Comment (Lansalo) [inline app/services/application/ApplicationService.scala]: In faststream though it was part of the story. Anyway... add it ;-)

**Pull request merged**